### PR TITLE
Lionelb/add elastic test script

### PR DESCRIPTION
**Pull request by lionelB**
add onogone test script.
for now test script output a report csv file but we need to generate a diff report so we can output it in PR.
ideas and comments are welcome !

### CODE REVIEW

**Comment by lionelB** [comment]
example below 
poke @revolunet  @douglasduteil 
---
```diff
    master     PR      +/-
==========================================
+   16.22%      38.34%      22.13%
==========================================
```
  
| file | Δ |  |
|---|---|---|
| rupture conventionnelle | `  100% <100%> (0%)` | :arrow_down: |
| rupture conventionnelle individuelle | `  100% <100%> (0%)` | :arrow_down: |
| rupture conventionnelle individuelle indemnité de rupture | `  0% <30%> (30%)` | :arrow_up: |
| rupture conventionnelle collective | `  0% <10%> (10%)` | :arrow_up: |
| rupture conventionnelle collective comment le contester ? | `  0% <20%> (20%)` | :arrow_up: |
| auto entrepreneur | `  0% <0%> (0%)` | :arrow_down: |
| lettre de demission | `  0% <30%> (30%)` | :arrow_up: |
| inspection du travail | `  0% <0%> (0%)` | :arrow_down: |
| cheque vacances | `  0% <40%> (40%)` | :arrow_up: |
| contrat de professionnalisation | `  0% <14%> (14%)` | :arrow_up: |
| CDI | `  0% <80%> (80%)` | :arrow_up: |
| dpae | `  0% <0%> (0%)` | :arrow_down: |
| cpf | `  100% <100%> (0%)` | :arrow_down: |
| sylae | `  0% <0%> (0%)` | :arrow_down: |
| cfe | `  0% <0%> (0%)` | :arrow_down: |
| chsct | `  0% <90%> (90%)` | :arrow_up: |
| vae | `  0% <0%> (0%)` | :arrow_down: |
| kbis | `  0% <0%> (0%)` | :arrow_down: |
| abandon de poste | `  0% <75%> (75%)` | :arrow_up: |
| congé parental | `  0% <0%> (0%)` | :arrow_down: |
| compte personnel de formation | `  100% <100%> (0%)` | :arrow_down: |
| solde de tout compte | `  0% <68%> (68%)` | :arrow_up: |
| licenciement économique conditions | `  0% <73.33%> (73.33%)` | :arrow_up: |
| congés payés | `  100% <100%> (0%)` | :arrow_down: |
| epargne salariale | `  0% <0%> (0%)` | :arrow_down: |
| harcèlement moral | `  0% <5%> (5%)` | :arrow_up: |
| certificat de travail | `  0% <33.33%> (33.33%)` | :arrow_up: |
| demission cdi chomage | `  0% <63.33%> (63.33%)` | :arrow_up: |
| licenciement pour faute grave | `  0% <93.33%> (93.33%)` | :arrow_up: |
| calcul indemnité rupture conventionnelle | `  0% <20%> (20%)` | :arrow_up: |
| contrat de professionnalisation salaire | `  0% <0%> (0%)` | :arrow_down: |
| formulaire accident du travail | `  0% <0%> (0%)` | :arrow_down: |
| combien est on payé en accident de travail | `  0% <0%> (0%)` | :arrow_down: |
| retraite anticipée travailleur handicapé fonction publique | `  0% <0%> (0%)` | :arrow_down: |
| licenciement économique | `  0% <40%> (40%)` | :arrow_up: |
| convention collective | `  100% <100%> (0%)` | :arrow_down: |
| congé maternité | `  0% <33.33%> (33.33%)` | :arrow_up: |

**Comment by lionelB** [comment]
@douglasduteil any idea why ci test fails 
> Cannot find module '@cdt/data...kali/utils' from 'lib.js'

maybe related, if I run `yarn lerna list`  
I get 0 packages found 🤔

**Comment by lionelB** [comment]
@douglasduteil can you review this PR, please ?

File: packages/code-du-travail-api/.gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (douglasduteil) [inline packages/code-du-travail-api/.gitignore]: ?

File: packages/code-du-travail-api/lib/__tests__/lib.test.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+});
+
+test("percent should return a percent value", () => {
+  const data = [
+    [0.1304323203230232, "13.04%"],
+    [0.1300323203230232, "13%"],
+    [0.0900323203230232, "9%"],
+    [0.0920323203230232, "9.2%"]
+  ];
+  data.forEach(([value, percentage]) => {
```
Comment (douglasduteil) [inline packages/code-du-travail-api/lib/__tests__/lib.test.js]: Can use `test.each` notation ?

File: packages/code-du-travail-api/lib/__tests__/lib.test.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      actual: ["A", "G", "F", "E"],
+      score: 0.25
+    },
+    {
+      expected: ["A", "", "", ""],
+      actual: ["A", "G", "F", "E"],
+      score: 1
+    }
+  ];
+  testCases.forEach(({ expected, actual, score }) => {
```
Comment (douglasduteil) [inline packages/code-du-travail-api/lib/__tests__/lib.test.js]: Can use `test.each` notation ?

File: packages/code-du-travail-api/lib/__tests__/lib.test.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    },
+    {
+      Expected_1: "e.url",
+      Expected_2: "d.url",
+      Expected_3: "c.url",
+      Expected_4: "b.url",
+      Expected_5: "a.url"
+    }
+  ];
+  testCases.forEach(line => {
```
Comment (douglasduteil) [inline packages/code-du-travail-api/lib/__tests__/lib.test.js]: Can use `test.each` notation ?

File: packages/code-du-travail-api/scripts/elastic.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  spinner.stop().clear();
+  const csvString = json2csv.parse(results, {
+    fields: Object.keys(results[0])
+  });
+  // fs.writeFileSync(csvFile, csvString);
+
+  // eslint-disable-next-line no-console
+  console.log(printResultsAbstract(results));
+  // eslint-disable-next-line no-console
+  console.log(printResultsDetails(results));
```
Comment (douglasduteil) [inline packages/code-du-travail-api/scripts/elastic.js]: Might want to pass console as argument of main.
'log' is a side effect.

File: packages/code-du-travail-api/src/server/routes/api.js [hunk 6]
```diff
@@ -18,6 +18,8 @@ const BASE_URL = `/api/v1`;
  */
 router.get(`${BASE_URL}/search`, async ctx => {
   let query = ctx.request.query.q;
+  let size = Math.min(ctx.request.query.size || 10, 100);
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/api.js]: Looks like a `const` to me...

**Review by douglasduteil — COMMENTED** [review]
- You might consider using file based testing as there are a lot of cases to tests.

File: packages/code-du-travail-api/lib/__tests__/lib.test.js (hunk 3 above)
Comment (lionelB) [reply packages/code-du-travail-api/lib/__tests__/lib.test.js]: Didn't know taht, thanks !

File: packages/code-du-travail-api/scripts/elastic.js (hunk 5 above)
Comment (lionelB) [reply packages/code-du-travail-api/scripts/elastic.js]: like ```main({reporter: console.log})  ```

and 
```
reporter(printResultsAbscract)
reporter(printResultsDetails ?)
```

File: packages/code-du-travail-api/src/server/routes/api.js (hunk 6 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/api.js]: indeed  😆

**Review by douglasduteil — APPROVED** [review]

**Pull request merged**